src/Microsoft.Net.Http.Headers/SetCookieHeaderValue.cs

```diff
@@ -342,15 +341,15 @@ private static int GetSetCookieLength(string input, int startIndex, out SetCooki
                     }
                     var dateString = ReadToSemicolonOrEnd(input, ref offset);
                     DateTimeOffset expirationDate;
-                    if (!HttpRuleParser.TryStringToDate(dateString, out expirationDate))
+                    if (!HttpRuleParser.TryStringToDate(dateString.ToString(), out expirationDate))
```

Comment:
HttpRuleParser is all internal, we can change its signature whenever. Leaving it makes it more obvious that we should revisit it, not that I expect to be writing a date parser from scratch any time soon.